File: levelledmobs-plugin-bukkit/src/main/java/io/github/arcaneplugins/levelledmobs/bukkit/logic/function/process/condition/impl/ChanceCondition.java
package io.github.arcaneplugins.levelledmobs.bukkit.logic.function.process.condition.impl;

import io.github.arcaneplugins.levelledmobs.bukkit.logic.context.Context;
import io.github.arcaneplugins.levelledmobs.bukkit.logic.function.process.Process;
import io.github.arcaneplugins.levelledmobs.bukkit.logic.function.process.condition.Condition;
import java.util.concurrent.ThreadLocalRandom;
import org.spongepowered.configurate.CommentedConfigurationNode;

public class ChanceCondition extends Condition {

    /* vars */

    private final float chance;

    /* constructors */

    public ChanceCondition(Process process, final CommentedConfigurationNode node) {
        super(process, node);
        this.chance = node.node("value").getFloat(100.0f);
    }

    /* methods */

    @Override
    public boolean applies(Context context) {
        final var random = ThreadLocalRandom.current().nextFloat();
        final var chance = (getChance() / 100.0f);
        return random <= chance;
    }

    /* getters and setters */

    public float getChance() { return chance; }

}

File: levelledmobs-plugin-bukkit/src/main/kotlin/io/github/arcaneplugins/levelledmobs/bukkit/logic/function/process/condition/impl/ChanceCondition.kt
package io.github.arcaneplugins.levelledmobs.bukkit.logic.function.process.condition.impl

import io.github.arcaneplugins.levelledmobs.bukkit.logic.context.Context
import io.github.arcaneplugins.levelledmobs.bukkit.logic.function.process.Process
import io.github.arcaneplugins.levelledmobs.bukkit.logic.function.process.condition.Condition
import org.spongepowered.configurate.CommentedConfigurationNode
import java.util.concurrent.ThreadLocalRandom

class ChanceCondition(
    process: Process,
    node: CommentedConfigurationNode
): Condition(process, node) {
    val chance = node.node("value").getFloat(100.0f)

    override fun applies(context: Context): Boolean {
        val random = ThreadLocalRandom.current().nextFloat()
        val chance: Float = chance / 100.0f
        return random <= chance
    }
}